packages/lesswrong/server/repos/CommentsRepo.ts

```diff
@@ -465,6 +466,126 @@ class CommentsRepo extends AbstractRepo<"Comments"> {
       WHERE "forumEventId" = $1 AND "userId" = $3
     `, [forumEventId, latestVote, userId]);
   }
+
+  /**
+   * Get comments for the UltraFeed
+   */
+  async getCommentsForFeed(
+    context: ResolverContext, 
+    maxTotalComments = 1000,
+    lookbackWindow = '90 days'
+  ): Promise<FeedCommentFromDb[]> {
+    const db = this.getRawDb();
+    const userId = context.userId;
+    const initialCandidateLimit = 100; // Limit for the first pass
+
+    const UNIVERSAL_COMMENT_FILTER_CLAUSE = `
+      c.deleted IS NOT TRUE
+      AND c.retracted IS NOT TRUE
+      AND c."authorIsUnreviewed" IS NOT TRUE
+      AND c."postId" IS NOT NULL
+    `;
+
+    const suggestedComments: FeedCommentFromDb[] = await db.manyOrNone(`
+      -- CommentsRepo.getCommentsForFeed
+      WITH
+      "CommentEvents" AS (
+        SELECT
+          "documentId",
+          MAX(CASE WHEN "eventType" = 'served' THEN "createdAt" END) AS "lastServed",
+          MAX(CASE WHEN "eventType" = 'viewed' THEN "createdAt" END) AS "lastViewed",
+          MAX(CASE WHEN "eventType" = 'expanded' THEN "createdAt" END) AS "lastInteracted"
+        FROM "UltraFeedEvents"
+        WHERE "userId" = $(userId)
+          AND "collectionName" = 'Comments'
+        GROUP BY "documentId"
+      ),
+      "InitialCandidates" AS (
+        SELECT
+          c._id AS "commentId",
+          COALESCE(c."topLevelCommentId", c._id) AS "threadTopLevelId",
+          c."postId",
+          CASE WHEN c."baseScore" > 10 THEN ARRAY['topComments'] ELSE ARRAY['quickTakes'] END AS sources
+        FROM "Comments" c
+        WHERE (c."baseScore" > 10 OR c.shortform IS TRUE)
+          AND ${UNIVERSAL_COMMENT_FILTER_CLAUSE}
+          AND ${getViewableCommentsSelector('c')}
+          AND c."postedAt" > NOW() - INTERVAL '${lookbackWindow}'
```

Comment:
I actually meant to inline the value of lookbackWindow.  (But if you're making it configurable, it needs to be via query parameter, not string interpolation.)

Reply:
It's an important parameter that I anticipate using elsewhere, also it's a pain to experiment with if it's buried in the query.